src/utils/checkName.ts

```diff
@@ -1,13 +1,47 @@
-const checkName = (name: string, nameList: string[]) => {
-  let nameCheck = "passCheck";
+interface IValidationOptions {
+  // if the name can be blank
+  canBeBlank?: boolean;
+  // maximum characters for the name
+  maxCharLength: number;
+  // minimum characters for the name
+  minCharLength?: number;
+  // if name contains only word characters (letter, number, space)
+  onlyWordChar?: boolean;
+}
+
+export const setValidation = (options: IValidationOptions) => {
+  const { canBeBlank = false, maxCharLength, minCharLength = 1, onlyWordChar = false } = options;
   const checkStartEndSpace = /(^\s*)|(\s*$)/g;
-  if (nameList.includes(name) || nameList.includes(name.replace(checkStartEndSpace, ""))) {
-    nameCheck = "existed";
-  }
-  if (name === "" || name.replace(checkStartEndSpace, "") === "") {
-    nameCheck = "blank";
-  }
-  // if () { nameCheck = "incorrect"; }
+  const onlyWordCharRegularExp = /^[a-zA-Z0-9\s]+$/g;
+  const validation = (name: string, nameList: string[]) => {
+    let result = "passCheck";
+    // check if name exists in nameList
+    if (nameList.includes(name) || nameList.includes(name.replace(checkStartEndSpace, ""))) {

```

Comment:
hi Jenna, a bit confused about this one, after checking AC we don't have to check the existing name,
but I found you write this function. and the  **const nameCheck = checkName(newDesignName, nameList, validation);
the nameList []** is blank, just double-check whether made this one or not on this ticket. thanks

Reply:
The feature of name existence checking was realised in the previous card. In this card, I refactored that part of code.
As for the nameList is blank, it is because the name list will only be updated when the user saves the court. It stands for the name list of saved courts by the current user.